src/date.rs

```diff
@@ -244,27 +240,49 @@ impl PartialOrd for HttpDate {
     }
 }
 
-/// Convert &[u8] to &str with zero checks.
-///
-/// For internal use only.
-/// Intended to be used with ASCII-only strings.
-fn conv(s: &[u8]) -> &str {
-    unsafe { from_utf8_unchecked(s) }
+fn toint_1(x: u8) -> Result<u8, Error> {
```

Comment:
Is this function (and the others) faster in the benchmarks than `u8::from_str`? By how much?

Reply:
As-is, using hand written toint:

```
parse_imf_fixdate       time:   [98.123 ns 99.939 ns 101.74 ns]                              
parse_rfc850_date       time:   [86.921 ns 87.186 ns 87.561 ns]                              
parse_asctime           time:   [96.105 ns 96.585 ns 97.099 ns]                          
```

When I replace all of them with 

```rust
fn toint_4(s: &[u8]) -> Result<u16, Error> {
    let s = std::str::from_utf8(&s).map_err(|_| Error(()))?;
    s.parse().map_err(|_| Error(()))
}
```

```
parse_imf_fixdate       time:   [166.66 ns 166.97 ns 167.40 ns]                              
                        change: [+69.318% +70.967% +72.504%] (p = 0.00 < 0.05)
                        Performance has regressed.

parse_rfc850_date       time:   [145.60 ns 145.95 ns 146.27 ns]                              
                        change: [+64.968% +65.975% +66.883%] (p = 0.00 < 0.05)
                        Performance has regressed.

parse_asctime           time:   [154.95 ns 154.99 ns 155.04 ns]                          
                        change: [+60.522% +61.152% +61.721%] (p = 0.00 < 0.05)
                        Performance has regressed.

```

This is *significantly* worse than the original numbers (which didn't have a UTF-8 check), and even those were beaten by this.